src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -47,9 +47,8 @@ public override void Initialize (AnalysisContext context)
 					ControlFlowGraph cfg = context.GetControlFlowGraph (operationBlock);
 					TrimDataFlowAnalysis trimDataFlowAnalysis = new (context, cfg);
 
-					foreach (TrimAnalysisPattern trimAnalysisPattern in trimDataFlowAnalysis.ComputeTrimAnalysisPatterns ()) {
-						foreach (var diagnostic in GetDynamicallyAccessedMembersDiagnostics (trimAnalysisPattern.Source, trimAnalysisPattern.Target, trimAnalysisPattern.Operation.Syntax.GetLocation ()))
-							context.ReportDiagnostic (diagnostic);
+					foreach (var diagnostic in trimDataFlowAnalysis.ComputeTrimAnalysisPatterns ().ReportDiagnostics ()) {
```

Comment:
Maybe we can rename `ReportDiagnostics` to make this read better - how about `CollectDiagnostics` or similar? I realize it might actually report things for the linker, but it would be nice to pick a more neutral name. I would also rename `DIagnosticContext.ReportDiagnostic` to `AddDiagnostic` which can mean either report for the linker or add to a collection for the analyzer.

Reply:
Which part of the linker do you mean? I don't see the changes - did you push to the PR? Thanks!

Reply:
There's nothing in the linker yet, but going forward the "patterns" is something we should be able to share, at which point this will be relevant.